src/Compilers/VisualBasic/Portable/Emit/SpecializedFieldReference.vb

```diff
@@ -70,12 +70,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
         End Function
 
         Private Sub AssociateWithMetadataWriter(metadataWriter As Cci.MetadataWriter) Implements Cci.IContextualNamedEntity.AssociateWithMetadataWriter
-            DirectCast(_underlyingField, Cci.IContextualNamedEntity).AssociateWithMetadataWriter(metadataWriter)
+            DirectCast(_underlyingField, SynthesizedStaticLocalBackingField).AssociateWithMetadataWriter(metadataWriter)
```

Comment:
>SynthesizedStaticLocalBackingField [](start = 41, length = 34)

Can this be reverted? It looks like `SynthesizedStaticLocalBackingField` implements `IContextualNamedEntity` and it's not obvious that the cast to `SynthesizedStaticLocalBackingField` will succeed.

Reply:
> Can this be reverted? It looks like SynthesizedStaticLocalBackingField implements IContextualNamedEntity and it's not obvious that the cast to SynthesizedStaticLocalBackingField will succeed.

```SynthesizedStaticLocalBackingField``` doesn't implement the interface in Release build. Also, the old cast succeeded only when ```_underlyingField``` was ```SynthesizedStaticLocalBackingField``` and it wasn't obvious it would succeed. I believe the consumer should first check  ```IsContextualNamedEntity```.


---
In reply to: [516287170](https://github.com/dotnet/roslyn/pull/49088#discussion_r516287170) [](ancestors = 516287170)